### DISCUSSION TITLE
Just running the installers

### DISCUSSION THREAD

**Discussion by Zhaph** [actions/virtual-environments#5581] [General questions] [answered]
Is there any issue with just running the installers on a given local machine to create an "equivalent" build agent? I'm not really looking to create a number of Azure hosted VMs nor VHDs - we have some servers available internally that I'd like to repurpose as build agents for our pipelines, and this would be a good first step in creating a solid baseline that we can work from to fully support the same set of features that we get from Hosted Build Agents.

**Comment by miketimofeev** [comment, marked as answer]
@Zhaph generally it is ok. We do it the same way when testing scripts during new OS version preparation.

**Reply by Zhaph** [reply]
Great news - I guess I'll need to run through the main image JSON files and pick out the bits I need, i.e. skip your configure AV, InitialiseVM, etc.

**Reply by miketimofeev** [reply]
correct, define some variables probably as well, shouldn't be too difficult
